servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -213,6 +234,16 @@ public void channelOutboundClosed() {
         promise.sourceTerminated(null);
     }
 
+    @Override
+    public void terminateSource() {
```

Comment:
can you qualify the method name as it seems very targeted toward expect/continue. I wonder if we could do something like `channelOutboundClosed(boolean expectationFailed)` to combine the two?